contentgrid-spring-data-rest/src/test/java/com/contentgrid/spring/data/rest/problem/ProblemDetailsMockMvcMatchers.java

```diff
@@ -0,0 +1,189 @@
+package com.contentgrid.spring.data.rest.problem;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import java.io.IOException;
+import java.net.URI;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.function.UnaryOperator;
+import java.util.stream.Stream;
+import lombok.AccessLevel;
+import lombok.AllArgsConstructor;
+import lombok.NoArgsConstructor;
+import lombok.RequiredArgsConstructor;
+import lombok.With;
+import org.assertj.core.api.ThrowingConsumer;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.HttpStatusCode;
+import org.springframework.http.ProblemDetail;
+import org.springframework.http.converter.json.ProblemDetailJacksonMixin;
+import org.springframework.test.web.servlet.MvcResult;
+import org.springframework.test.web.servlet.ResultMatcher;
+
+@NoArgsConstructor(access = AccessLevel.PRIVATE)
+public final class ProblemDetailsMockMvcMatchers {
+
+    private final static ObjectMapper objectMapper = new ObjectMapper();
+
+    static {
+        objectMapper.addMixIn(ProblemDetail.class, ProblemDetailJacksonMixin.class);
+    }
+
+    public static ProblemDetailsMatcher problemDetails() {
+        return new ProblemDetailsMatcher();
+    }
+
+    public static ValidationConstraintViolationMatcher validationConstraintViolation() {
+        return new ValidationConstraintViolationMatcher(List.of());
+    }
+
+    @With
+    @AllArgsConstructor
+    public static class ProblemDetailsMatcher implements ResultMatcher {
+
+        private final static String SENTINEL = "\0";
```

Comment:
what ? why ?

Reply:
This is a matcher used for tests.

It needs to distinguish between:

- Value not configured (don't perform a match) -> `SENTINEL`
- Value configured to null (match to null/absence of value) -> `null`
- Value configured to a string (match to that string)

We need some kind of value that can be used for the first case.
The string `\0` is very unlikely to be used in any problem details response, so I used that.

Reply:
So there was a semi-legitimate use for :hankey: and you did not go for that ? I'm disappointed now.